### ISSUE TITLE
Idea: allow the fire escape key to be interacted with without the flashlight

### ISSUE THREAD

**Issue by matheusvb3** [elishacloud/Silent-Hill-2-Enhancements#838]
This is something I've been thinking about since [this issue](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/748) was written. In the Wood Side apartments there are three conditions that have to be met for the corpse to appear in room 208:
1. The player must have the flaslight
2. The player must have the handgun
3. The player must have watched the cutscene where Laura kicks the fire escape key away from James' hands.

There are two orders in which the player can go about accomplishing this:
- Get flashlight, get gun, watch cutscene
- Get flashlight, watch cutscene, get gun

However, there is no real reason for the game to restrict the player from watching the cutscene _without_ the flashlight. The key that James must get is being lit by a source coming from the laundry area in the third floor, and it's already very visible even without the flashlight. James even _**looks**_ at the key when the player enters the hallway, and if you get close to the bars the camera switches to give a better view of the key on the floor, yet you can't do anything about it. This actually caused me some confusion when I went to play this game after a long time since my last playthrough, and I was left wondering why he was _looking_ at the key but I couldn't interact with it, until I remembered "oh yeah, the flashlight."

The apartments are already the most "item-hunt" part of the game, being very strict with how you go about acquiring the items and bringing them to their correct locations, and I'd even say they're the worst designed level (so to speak) in the game. By allowing the player to watch the cutscene without having the flashlight, now there are three ways of accomplishing the requirements for the corpse to appear in room 208 and progress further:
- Get flashlight, get gun, watch cutscene
- Get flashlight, watch cutscene, get gun
- _Watch cutscene, get flashlight, get gun_

I'm sure there are other areas in the game where James will refuse to interact with an item because he has no flashlight even if it's being lit by another source (shoutouts to Lakeview Hotel), but this is just one change that I think could greatly benefit this game.

Yes, I know this mod doesn't want to alter how the game needs to be played and is intended to fix the issues from the porting process, but I plead some consideration is taken about this change. It's something I think would genuinely make the game better, not only for first time players but everyone who replays it and is left wondering why James is being stubborn all of a sudden.

The following images are captured on PCSX2 (since that's the version the mod is trying to have the most fidelity to, graphics and presentation-wise), with the ingame brightness setting on the lowest value and noise grain enabled:

![key](https://github.com/elishacloud/Silent-Hill-2-Enhancements/assets/98937378/48582c22-94d2-4ce5-8971-f1db88b1fe50)
![key2](https://github.com/elishacloud/Silent-Hill-2-Enhancements/assets/98937378/68e16e1d-7add-4129-9f4c-36ab0ba0a2ab)

Notice how in the first image you can clearly see the key on the ground because of its specularity effect.

**Comment by Polymega** [comment]
This is an excellent suggestion for something that has bothered me _for years._ And as you said, I wouldn't be as bothered by this _if James wasn't already looking at the key **before** acquiring the flashlight._ I also fully agree that this whole thing hurts more than helps with player confusion.

It might be worth looking into. My only fear is how this would break things. It might break the "chain of events" the game is expecting during this part. For example: Would this break the sequence of the man screaming in the hallways?

> Yes, I know this mod doesn't want to alter how the game needs to be played and is intended to fix the issues from the porting process, but I plead some consideration is taken about this change. It's something I think would genuinely make the game better, not only for first time players but everyone who replays it and is left wondering why James is being stubborn all of a sudden.

[We have no problem adjusting things in the game if it makes for a better user experience.](https://youtu.be/LylafxQta9w?si=5ioiE5BkkGDb5_Nf&t=1139) Our goal with this project is _not_ to be 1:1 with the PS2 version in everything, but rather restore _and_ improve things to make a great, remastered experience for the game.

**Comment by matheusvb3** [comment]
My belief is that, from what we can infer playing the game normally, for the "man screaming" trigger it checks the "watched cutscene" flag to see if it's set to true/has a value of 1 and checks the player's inventory for the flashlight and handgun, so I wouldn't expect it to really care _what order_ these conditions are met. What could be problematic is getting the flashlight _while_ the "watched cutscene" flag is active as that's impossible to do normally, though I have a feeling it won't cause issues. It could be tested by hex editing a PCSX2 save file and forcing the "watched cutscene" flag to be true and then obtaining the flashlight (it could also be done on the PC version but the save files are encrypted...), then playing the rest of the game to make extra sure it doesn't have any cascading effects. If it proves to not be problematic, then the next challenge would be finding and forcibly removing the flashlight check the developers put in place for interacting with the key, I believe.

**Comment by Polymega** [comment]
I Just now discovered/realized: To further justify this case, James will comment about the key _without_ his flashlight on after the fact:

![Screenshot 2023-09-08 10-21-52](https://github.com/elishacloud/Silent-Hill-2-Enhancements/assets/27395260/7e3575e0-7a4f-4e3e-bd39-8b3127828735)

**Comment by AeroWidescreen** [comment]
@elishacloud 
All you need to do is write a byte of 0x00 at the follow memory addresses:

Examine Fire Escape Key (Behind Bars)
1.0: 008E1294
1.1: 008E4F64
DC: 008E3F64

Pickup Fire Escape Key
1.0: 008E0ABC
1.1: 008E478C
DC: 008E378C

Pointers for 1.0:
00494F02 -> (Offset: 0x00) 008E0DF0 -> (Offset: 0x83C) 008E1294
00494F02 -> (Offset: 0x00) 008E0DF0 -> (Offset: 0x64) 008E0ABC

Changing the original value from 0x04 to 0x00 will allow users to interact with this object without needing the flashlight. There doesn't appear to be any negative side-effects by doing this. Players will still be required to acquire the flashlight and weapon if they want to trigger the scream and proceed through the level as they normally would.

**Comment by matheusvb3** [comment]
> I Just now discovered/realized: To further justify this case, James will comment about the key _without_ his flashlight on after the fact:

You can also use the map when standing in this spot, you have to walk about 7 feet away from the bars until the "it's too dark to read the map" message gets triggered.

This seems to indicate the developers knew the area next to the bars was too bright, so they deliberately decided to put a check in place to force players to get the flashlight before watching the cutscene.

**Comment by matheusvb3** [comment]
I just realized I mixed up the key James has to get, it's actually the fire escape one. Major oops.

**Comment by elishacloud** [comment]
I put a fix in for this based on @AeroWidescreen's comments above.  I am not quite clear on when the value needs to change from 0x04 to 0x00.  I tried setting it at launch and it made the SH2 window appear in the corner.

For now, I just set this value once during the first frame after the game starts up.  I am not sure if I ever need to set the value again or not.  Let me know if this works.

Here is a test build:  [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/13255650/d3d8.zip)
